PhysicsTools/NanoAOD/python/taus_cff.py

```diff
@@ -175,12 +175,12 @@ def _tauId8WPMask(pattern,doc):
                  idAntiEleDeadECal = Var("tauID('againstElectronDeadECALForNano')", bool, doc = "Anti-electron dead-ECal discriminator"),
     )
 
-tauGenJets.GenParticles = cms.InputTag("prunedGenParticles")
-tauGenJets.includeNeutrinos = cms.bool(False)
+tauGenJets.GenParticles = "finalGenParticles"
+tauGenJets.includeNeutrinos = False
 
 genVisTaus = cms.EDProducer("GenVisTauProducer",
     src = cms.InputTag("tauGenJetsSelectorAllHadrons"),         
-    srcGenParticles = cms.InputTag("prunedGenParticles")
+    srcGenParticles = cms.InputTag("finalGenParticles")
```

Comment:
you can remove here as well the `cms.InputTag("` ?

Reply:
No, it will be incorrect. In case of `tauGenJets` arguments of already defined EDProducer module are modifier which is better to do by changing its value without its type specification, while here a new EDProducer is defined (`genVisTaus = cms.EDProducer("GenVisTauProducer"`) so types of inputs have to be explicitly provided.